### PR TITLE
MSI-2619: Stock status for sold out product don't change to 'Out of stock"

### PR DESCRIPTION
**Pull request by nmalevanec**
### Description (*)

Product gets 'Out of Stock' status on specific source after source quantity becomes equal to 'Out of Stock Threshold'.

### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2619: Stock status for sold out product don't change to 'Out of stock"

### Manual testing scenarios (*)

1. see https://github.com/magento/inventory/issues/2619

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/InventorySourceDeductionApi/Model/SourceDeductionService.php [hunk 1]
```diff
@@ -93,4 +100,21 @@ public function execute(SourceDeductionRequestInterface $sourceDeductionRequest)
...
+     *
+     * @param StockItemConfigurationInterface $stockItemConfiguration
+     * @param SourceItemInterface $sourceItem
+     * @return int
+     */
+    private function getSourceStockStatus(
+        StockItemConfigurationInterface $stockItemConfiguration,
+        SourceItemInterface $sourceItem
+    ): int {
+        $sourceItemQty = $sourceItem->getQuantity() ?: 0.0;
```
Comment (Stepa4man) [inline app/code/Magento/InventorySourceDeductionApi/Model/SourceDeductionService.php]: Do you think it makes sense to use private constant here?

**Review by Stepa4man — CHANGES_REQUESTED** [review]
Hi @nmalevanec 

Thanks for your contribution.
Could you please check my comments and integration tests?

File: app/code/Magento/InventorySourceDeductionApi/Model/SourceDeductionService.php (hunk 1 above)
Comment (nmalevanec) [reply app/code/Magento/InventorySourceDeductionApi/Model/SourceDeductionService.php]: Done

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento run Functional Tests

**Pull request merged**